src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -172,7 +172,10 @@ class DetailsCard extends React.Component {
     const vmClusterId = this.props.vm.getIn(['cluster', 'id'])
 
     if (prevProps.clusters !== this.props.clusters) {
-      this.setState({ clusterList: createClusterList(this.props.clusters) }) // eslint-disable-line react/no-did-update-set-state
+      const vmClusterId = this.props.vm.getIn(['cluster', 'id'])
+      const vmDataCenterId = this.props.clusters.getIn([vmClusterId, 'dataCenterId'])
+      const { clusters, hosts } = this.props
```

Comment:
Ben, is there any purpose for the order of the two lines above? I think you could use
```javascript
      const { clusters, hosts } = this.props
      const vmDataCenterId = clusters.getIn([vmClusterId, 'dataCenterId'])
```
instead.

Also I think you don't need the line 175 `const vmClusterId = this.props.vm.getIn(['cluster', 'id'])`
See the line 172 above the "if" block.